backend/src/govsg/services/govsg-callback.service.ts

```diff
@@ -88,8 +103,45 @@ const parseWebhook = async (
   throw new UnexpectedWebhookError('Unexpected webhook body')
 }
 
+const sendPasscodeCreationMessage = async (
+  whatsappId: WhatsAppId,
+  clientId: WhatsAppApiClient
+): Promise<MessageId> => {
+  const templateMessageToSend: WhatsAppTemplateMessageToSend = {
+    recipient: whatsappId,
+    apiClient: clientId,
+    templateName: 'sgc_passcode_generation', // TODO: Un-hardcode this
+    params: [],
+    language: WhatsAppLanguages.english,
+  }
+  const isLocal = config.get('env') === 'development'
+  const passcodeCreationWamid =
+    await WhatsAppService.whatsappClient.sendTemplateMessage(
+      templateMessageToSend,
+      isLocal
+    )
+  return passcodeCreationWamid
+}
+
+const createPasscode = () => {
+  return randomInt(0, Math.pow(10, 4)).toString().padStart(4, '0')
+}
+
+const storePrecreatedPasscode = async (
```

Comment:
Could I ask why we consider this precreated? Doesn't this value get generated on demand? (And thus we can supply `passcodeCreationWamid` as a param)?

Reply:
It's pre-created in our perspective because `sgc_passcode_generation` is the template that comes with the "Create passcode" button. The user has not told us that he/she is creating a passcode yet, but we already have it pre-created.

Reply:
So what you mean is that we create the passcode as soon as the message is sent, instead of when the user clicks on the button?

Reply:
Yes!